### PR TITLE
Added: `selector-max-combinators` rule.

### PR DESCRIPTION
**Pull request by evilebottnawi**
> Which issue, if any, is this issue related to?

#2528

> Is there anything in the PR that needs further explanation?

No, it's self explanatory.

### CODE REVIEW

File: docs/user-guide/rules.md [hunk 1]
```diff
@@ -163,6 +163,7 @@ Here are all the rules within stylelint, grouped by the [*thing*](http://apps.wo
 -   [`selector-descendant-combinator-no-non-space`](../../lib/rules/selector-descendant-combinator-no-non-space/README.md): Disallow non-space characters for descendant combinators of selectors.
 -   [`selector-id-pattern`](../../lib/rules/selector-id-pattern/README.md): Specify a pattern for id selectors.
 -   [`selector-max-class`](../../lib/rules/selector-max-class/README.md): Limit the number of classes in a selector.
+-   [`selector-max-combinators`](../../lib/rules/selector-max-combinators/README.md): Limit the number of combinators selectors in a selector.
```
Comment (jeddy3) [inline docs/user-guide/rules.md]: "Limit the number of combinators in a selector."

File: lib/rules/selector-max-combinators/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# selector-max-combinators
+
+Limit the number of combinators selectors in a selector.
```
Comment (jeddy3) [inline lib/rules/selector-max-combinators/README.md]: "Limit the number of combinators in a selector."

File: docs/user-guide/rules.md (hunk 1 above)
Comment (evilebottnawi) [inline docs/user-guide/rules.md]: @jeddy3 done

File: lib/rules/selector-max-combinators/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+```css
+a {}
+```
+
+```css
+a b {}
+```
+
+```css
+a, b {}
```
Comment (jeddy3) [inline lib/rules/selector-max-combinators/README.md]: Use `a b ~ c` here for parity with positive patterns.

File: lib/rules/selector-max-combinators/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+```
+
+```css
+a.foo {}
```
Comment (jeddy3) [inline lib/rules/selector-max-combinators/README.md]: Use the following instead here:

```css
a b {
  & ~ c {}
}
```

File: lib/rules/selector-max-combinators/README.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+```
+
+```css
+/* each selector in a selector list is evaluated separately */
```
Comment (jeddy3) [inline lib/rules/selector-max-combinators/README.md]: Use the following instead: 

```css
a b,
c > d {}
```

File: lib/rules/selector-max-combinators/index.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+const parseSelector = require("../../utils/parseSelector")
+const report = require("../../utils/report")
+const ruleMessages = require("../../utils/ruleMessages")
+const validateOptions = require("../../utils/validateOptions")
+const resolvedNestedSelector = require("postcss-resolve-nested-selector")
+
+const ruleName = "selector-max-combinators"
+
+const messages = ruleMessages(ruleName, {
+  expected: (selector, max) => `Expected "${selector}" to have no more than ${max} combinators ${max === 1 ? "selector" : "selectors"}`,
```
Comment (jeddy3) [inline lib/rules/selector-max-combinators/index.js]:
```js
expected: (selector, max) => `Expected "${selector}" to have no more than ${max}  ${max === 1 ? "combinator" : "combinators"}`,
```

**Comment by evilebottnawi** [comment]
@jeddy3 done, about tests: all tested was migrate from basic rules and add more tests, in theory all should be fine. Btw, if someone found error we just fix it and release patch version.

File: lib/rules/selector-max-combinators/__tests__/index.js [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+
+  accept: [ {
+    code: "foo bar {}",
+    description: "fewer than max id selectors",
+  }, {
+    code: "foo bar:hover {}",
+    description: "pseudo selectors",
+  }, {
+    code: "foo bar, \nbaz quux {}",
+    description: "multiple selectors: exactly max id selectors",
```
Comment (hudochenkov) [inline lib/rules/selector-max-combinators/__tests__/index.js]: “multiple selectors: fewer then max combinators selectors”

**Comment by jeddy3** [comment]
@hudochenkov Changes made.

**Comment by jeddy3** [comment]
> -   Added: `selector-max-combinators` rule ([#2658](https://github.com/stylelint/stylelint/pull/2658)).

**Pull request merged**